File: backend/src/main/java/pers/eastwind/billmanager/company/repository/CompanyRepository.java
package pers.eastwind.billmanager.company.repository;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Repository;
import pers.eastwind.billmanager.common.repository.BaseRepository;
import pers.eastwind.billmanager.company.model.Company;

import java.util.List;

/**
 * 公司 Repository
 */
@Repository
public interface CompanyRepository extends BaseRepository<Company> {
    Page<Company> findByIsDisabled(boolean isDisabled, Pageable pageable);
    List<Company> findByNameContains(String name);
}

File: backend-go/internal/company/repository/company_repository.go
package repository

import (
	"backend-go/internal/common/database"
	"backend-go/internal/common/errs"
	"backend-go/internal/common/result"
	"backend-go/internal/company/model"
	"context"

	"gorm.io/gorm"
)

type CompanyRepository struct {
	*database.BaseRepository[model.Company]
}

func NewCompanyRepository(db *gorm.DB) *CompanyRepository {
	return &CompanyRepository{
		BaseRepository: database.NewBaseRepository[model.Company](db),
	}
}

// FindAllEnabled 查找所有启用的公司
func (r *CompanyRepository) FindAllEnabled(ctx context.Context, pageable *result.QueryParam) (*result.PageResult[model.Company], errs.StackError) {
	return r.FindAllWithPage(ctx, pageable, "is_disabled = ?", false)
}

// FindEnabledByNameContains 根据名称查找启用的公司
func (r *CompanyRepository) FindEnabledByNameContains(ctx context.Context, name string) (*[]model.Company, errs.StackError) {
	return r.FindAll(ctx, "is_disabled = ? and name LIKE ?", false, "%"+name+"%")
}
